### PR TITLE
Fix dependency version & packageVersion in runtime.native.system.io.ports

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/runtime/issues/47268

# Description
We shipped runtime.native.system.io.ports package with non-stable dependencies

 

```

runtime.linux-arm.runtime.native.System.IO.Ports (>= 5.0.0-rtm.20519.4)
runtime.linux-arm64.runtime.native.System.IO.Ports (>= 5.0.0-rtm.20519.4)
runtime.linux-x64.runtime.native.System.IO.Ports (>= 5.0.0-rtm.20519.4)
runtime.osx-x64.runtime.native.System.IO.Ports (>= 5.0.0-rtm.20519.4)

```

 

Nuget.exe fails to install the package with specific targetFramework

 

eg. NuGet.exe install System.IO.Ports -Version 5.0.0 -outputdirectory ioports -Framework net5.0 -Verbosity detailed

 

 

We also never shipped the stable 5.0 version

```

runtime.linux-arm.runtime.native.System.IO.Ports (>= 5.0.0-rtm.20519.4)
runtime.linux-arm64.runtime.native.System.IO.Ports (>= 5.0.0-rtm.20519.4)
runtime.linux-x64.runtime.native.System.IO.Ports (>= 5.0.0-rtm.20519.4)
runtime.osx-x64.runtime.native.System.IO.Ports (>= 5.0.0-rtm.20519.4)

```
of these packages.

The fix is shipping 5.0.0 stable version of runtime specific  packages and 5.0.1 version of  runtime.native.system.io.ports with 5.0.0 dependencies 

# Customer Impact
Nuget pack does not allow to have a stable package depend on non-stable packages. 
Customers lack confidence in our packages due to non-stable suffix. There wont be any functional change.

# Regression

yes, in 3.1 we shipped the stable versions of runtime specific packages 

# Testing

Used package explorer to check the dependency versions. 
Verified with nuget.exe install works using a local feed.

# Risk

 Low, we are changing the packageIndex.json to fix this. The major use of PackageIndex is used to calculate the dependency version in packages. However, the change is very specific to few packages.

### CODE REVIEW

**Review by ViktorHofer — APPROVED** [review]
Is this scenario broken in master as well?

File: src/libraries/pkg/baseline/packageIndex.json [hunk 1]
```diff
@@ -1506,11 +1506,40 @@
...
+      "InboxOn": {}
+    },
+    "runtime.linux-x64.runtime.native.System.IO.Ports": {
+      "StableVersions": [
+        "4.7.0",
+        "5.0.0"
+      ],
+      "InboxOn": {}
+    },
+    "runtime.osx-x64.runtime.native.System.IO.Ports": {
```
Comment (safern) [inline src/libraries/pkg/baseline/packageIndex.json:1536]: What happens if we add new rids, do we have to be adding more entries here? I don't know if there is a better way long term. For 5.0.0 fix, it seems good as is, but maybe we can come with a better way long term.

File: src/libraries/libraries-packages.proj [hunk 2]
```diff
@@ -18,6 +18,7 @@
     <ProjectReference Include="$(PkgDir)*\*.proj" Exclude="$(PkgDir)test\*" Condition="'$(BuildAllOOBPackages)' == 'true'" />
     <ProjectReference Include="$(MSBuildThisFileDirectory)*\pkg\**\*.pkgproj" Condition="('$(BuildAllConfigurations)' == 'true' or '$(DotNetBuildFromSource)' == 'true') And '$(BuildAllOOBPackages)' == 'true'" />
     <!-- If setting BuildAllOOBPackages to false, add bellow the individual OOB packages you want to continue to build -->
+        <ProjectReference Include="$(PkgDir)runtime.native.System.IO.Ports\runtime.native.System.IO.Ports.proj" />
```
Comment (ericstj) [inline src/libraries/libraries-packages.proj:21]: nit: indentation

Comment (ericstj) [reply src/libraries/libraries-packages.proj:21]: Shouldn't we also build System.IO.Ports so that it gest a reference to the new runtime.System.IO.Ports?

**Review by ericstj — CHANGES_REQUESTED** [review]
Please ensure we build a new version of System.IO.Ports so folks have a public package to reference to get this fix.

**Comment by danmosemsft** [comment]
taking label off until it's ready to go

File: src/libraries/pkg/baseline/packageIndex.json (hunk 1 above)
Comment (ericstj) [reply src/libraries/pkg/baseline/packageIndex.json:1536]: In 6.0 the package index will go away.

**Comment by danmosemsft** [comment]
Also @Anipik can you please update the template content above to your improved version that is clearer eg about the customer impact of the current state.

**Comment by ericstj** [comment]
Approved over mail

File: src/libraries/System.IO.Ports/Directory.Build.props [hunk 3]
```diff
@@ -4,5 +4,6 @@
     <StrongNameKeyId>Open</StrongNameKeyId>
     <IncludePlatformAttributes>true</IncludePlatformAttributes>
     <UnsupportedOSPlatforms>browser</UnsupportedOSPlatforms>
+    <PackageVersion>5.0.1</PackageVersion>
```
Comment (ericstj) [inline src/libraries/System.IO.Ports/Directory.Build.props:7]: Given no change to the assembly I am OK without AssemblyVersion bump here, but normally we'd bump it.

**Review by ericstj — APPROVED** [review]

File: src/libraries/System.IO.Ports/Directory.Build.props (hunk 3 above)
Comment (Anipik) [reply src/libraries/System.IO.Ports/Directory.Build.props:7]: > Given no change to the assembly I am OK without AssemblyVersion bump here, but normally we'd bump i

Thats why kept it the same.

**Comment by Anipik** [comment]
Failures are related to this https://github.com/dotnet/runtime/issues/46837

**Pull request merged**